### DISCUSSION TITLE
how to obsfucate source code in production

### DISCUSSION THREAD

**Discussion by deschantkn** [vercel/next.js#27684] [Help]
my Next.js app's source is readable in production by inspecting sources.

is there a way to obfuscate the source code?

**Comment by lukecaan** [comment]
Any chance you got an answer to this?

**Reply by deschantkn** [reply]
Ended up finding a webpack plugin https://github.com/javascript-obfuscator/webpack-obfuscator. It solved my problem

**Reply by lukecaan** [reply]
Cheers!

**Reply by andrienpecson** [reply]
@deschantkn do you have an example of implementing https://github.com/javascript-obfuscator/webpack-obfuscator to next.js?

**Reply by hishamank** [reply]
@andrienpecson  you need to install `webpack-obfuscator` and `javascript-obfuscator`
and in your `next.config.js`

add this import 

`const webpackObfuscator = require('webpack-obfuscator');`

and add this to your webpack configs

```
  webpack: (config, { isServer }) => {
    if (!isServer) {
      config.module.rules.push({
        test: /\.(ts|tsx)$/,
        exclude: /node_modules/,
        enforce: 'post',
        use: [
          {
            loader: webpackObfuscator.loader,
            options: {
              compact: true,
              .........
            }
            
            
          }
        ]
      });
    }
    return config;
  }
```
  
replace ..... with your options, 
options can be found in https://github.com/javascript-obfuscator/javascript-obfuscator 
find the ones that meet your requirements

**Reply by kundanstudent11** [reply]
> @andrienpecson you need to install `webpack-obfuscator` and `javascript-obfuscator` and in your `next.config.js`
> 
> add this import
> 
> `const webpackObfuscator = require('webpack-obfuscator');`
> 
> and add this to your webpack configs
> 
> ```
>   webpack: (config, { isServer }) => {
>     if (!isServer) {
>       config.module.rules.push({
>         test: /\.(ts|tsx)$/,
>         exclude: /node_modules/,
>         enforce: 'post',
>         use: [
>           {
>             loader: webpackObfuscator.loader,
>             options: {
>               compact: true,
>               .........
>             }
>             
>             
>           }
>         ]
>       });
>     }
>     return config;
>   }
> ```
> 
> replace ..... with your options, options can be found in https://github.com/javascript-obfuscator/javascript-obfuscator find the ones that meet your requirements


After implement this, env value is not loaded
